File: sorting/sequencial/go/bubble_sort.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
	"time"
)

func bubble_sort(ary []int) []int {
	for n := size - 1; n > 0; n-- {
		for i := 0; i < n; i++ {
			if ary[i] > ary[i+1] {
				tmp := ary[i]
				ary[i] = ary[i+1]
				ary[i+1] = tmp
			}
		}
	}
	// fmt.Println(ary)
	return ary
}

var (
	size = 100000
)

func main() {
	fp, _ := os.Open("../../../data/" + strconv.Itoa(size) + "/seed0.txt")
	scanner := bufio.NewScanner(fp)
	ary1 := make([]int, size)
	i := 0
	for scanner.Scan() {
		ary1[i], _ = strconv.Atoi(scanner.Text())
		i++
	}
	fp, _ = os.Open("../../../data/" + strconv.Itoa(size) + "/seed1.txt")
	scanner = bufio.NewScanner(fp)
	ary2 := make([]int, size)
	i = 0
	for scanner.Scan() {
		ary2[i], _ = strconv.Atoi(scanner.Text())
		i++
	}
	fp, _ = os.Open("../../../data/" + strconv.Itoa(size) + "/seed2.txt")
	scanner = bufio.NewScanner(fp)
	ary3 := make([]int, size)
	i = 0
	for scanner.Scan() {
		ary3[i], _ = strconv.Atoi(scanner.Text())
		i++
	}

	sTime := time.Now()
	bubble_sort(ary1)
	bubble_sort(ary2)
	bubble_sort(ary3)
	eTime := time.Now()
	fmt.Println(eTime.Sub(sTime) / 3)
}

File: sorting/sequencial/ruby/bubble_sort.rb
require 'benchmark'

def bubble_sort(ary)
  n = ary.length
  (n - 1).times do |k|
    (n - k - 1).times do |i|
      if ary[i] > ary[i + 1]
        tmp = ary[i]
        ary[i] = ary[i + 1]
        ary[i + 1] = tmp
      end
    end
  end
  ary
end

size = 10000
f1 = File.open("./../../data/#{size}/seed0.txt", 'r')
ary1 = Array.new(size)
index = 0
f1.each_line do |row|
  ary1[index] = row.chomp.to_i
  index += 1
end
f2 = File.open("./../../data/#{size}/seed1.txt", 'r')
ary2 = Array.new(size)
index = 0
f2.each_line do |row|
  ary2[index] = row.chomp.to_i
  index += 1
end
f3 = File.open("./../../data/#{size}/seed2.txt", 'r')
ary3 = Array.new(size)
index = 0
f3.each_line do |row|
  ary3[index] = row.chomp.to_i
  index += 1
end

result = Benchmark.realtime do
  bubble_sort(ary1)
  bubble_sort(ary2)
  bubble_sort(ary3)
end
p result / 3
